engine/src/org/pentaho/di/repository/kdr/KettleDatabaseRepository.java

```diff
@@ -2017,6 +2017,13 @@ public RepositoryObject getObjectInformation( ObjectId objectId, RepositoryObjec
           dirId = row.getInteger( KettleDatabaseRepository.FIELD_JOB_ID_DIRECTORY, 0 );
           break;
         }
+        //PDI-15871 Return available information for DATABASE
+        case DATABASE: {
+          RowMetaAndData row = databaseDelegate.getDatabase( objectId );
+          name = row.getString( KettleDatabaseRepository.FIELD_DATABASE_NAME, null );
+          return new RepositoryObject(
+              objectId, name, null, null, null, objectType, null, false );
+        }
         default:
           throw new KettleException( "Object type "
             + objectType.getTypeDescription()
```

Comment:
You need to correct the message if you are also going to return information about type DATABASE